src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -436,33 +433,6 @@ export class Kernel implements IKernel {
             });
         }
     }
-    private async initializeMatplotlib(): Promise<void> {
```

Comment:
This init code was running in two places. Also, the code was similar but different enough that it was actually getting different results from the two different calls so the results were conficting in the notebook. Locking on the notebook version for now as I believe that is correct for as long as we are still using INotebook, seems related to the refresh discussion as well.

Reply:
> n the notebook. Locking on the notebook version for now as I believe that is correct for as long as we are still us

Please can we move the initialization to `Kernel`, as the plan is to move all of INotebook initialization into this.

Reply:
@DonJayamanne moved the init code (working dir / startup commands / plot init) into kernel.